Ast/src/Lexer.cpp

```diff
@@ -535,37 +587,67 @@ Lexeme Lexer::readQuotedString()
             return Lexeme(Location(start, position()), Lexeme::BrokenString);
 
         case '\\':
+            readBackslashInString();
+            break;
+
+        default:
             consume();
-            switch (peekch())
-            {
-            case '\r':
-                consume();
-                if (peekch() == '\n')
-                    consume();
-                break;
+        }
+    }
 
-            case 0:
-                break;
+    consume();
 
-            case 'z':
-                consume();
-                while (isSpace(peekch()))
-                    consume();
-                break;
+    return Lexeme(Location(start, position()), Lexeme::QuotedString, &buffer[startOffset], offset - startOffset - 1);
+}
 
-            default:
-                consume();
-            }
+Lexeme Lexer::readInterpolatedStringBegin()
+{
+    LUAU_ASSERT(peekch() == '`');
+
+    Position start = position();
+    consume();
+
+    return readInterpolatedStringSection(start, Lexeme::InterpStringBegin, Lexeme::InterpStringSimple);
+}
+
+Lexeme Lexer::readInterpolatedStringSection(Position start, Lexeme::Type formatType, Lexeme::Type endType)
+{
+    unsigned int startOffset = offset;
+
+    while (peekch() != '`')
+    {
+        switch (peekch())
+        {
+        case 0:
+        case '\r':
+        case '\n':
+            // INTERP TODO: Clear anything we've added to the brace stack, and write a test to see what happens if we don't
+            return Lexeme(Location(start, position()), Lexeme::BrokenString);
+
+        case '\\':
+            readBackslashInString();
             break;
 
+        case '{':
```

Comment:
So I see this scans from `` ` `` until `{`. We need to consume `\u{` first before this one, otherwise `` `\u{42}` `` will be equivalent to `"u42"` instead of `"A"`